pkg/module/remote.go

```diff
@@ -126,45 +125,47 @@ func NoSchemeURL(url string) string {
 	return regex.ReplaceAllString(url, "")
 }
 
-// Push picks up the archive described in the config and pushes it to the provided registry.
+// Push picks up the archive described in the config and pushes it to the provided registry if not existing.
```

Comment:
I think this entire godoc deserves some improvement.
Maybe we should also change the method name to reflect it's current logic? Something like `PushIfNotExists` ?
Another idea would be to keep the `Push` method for pushing - without any conditions, and move all the resources comparison to another method.
To summarize: I think this method now does too much, it's hard to even write a good docs for it.